src/Syntax/Parse.hs

```diff
@@ -60,12 +61,13 @@ import Syntax.Lexeme
 import Syntax.Lexer   ( lexing )
 import Syntax.Layout  ( layout )
 import Syntax.Promote ( promote, promoteType, quantify, promoteFree )
+import Common.ColorScheme (defaultColorScheme)
 
 -----------------------------------------------------------
 -- Parser on token stream
 -----------------------------------------------------------
 
-type LexParser a  = Parsec [Lexeme] () a -- GenParser Lexeme () a
+type LexParser a  = Parsec [Lexeme] [(String, Range)] a -- GenParser Lexeme () a
```

Comment:
The main change here is to report syntax warnings in batch, and not cause termination. This way we can report syntax errors in the language server.